src/v/kafka/server/tests/metadata_test.cc

```diff
@@ -0,0 +1,336 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "cluster/config_frontend.h"
+#include "cluster/security_frontend.h"
+#include "kafka/client/transport.h"
+#include "kafka/protocol/create_topics.h"
+#include "kafka/protocol/metadata.h"
+#include "kafka/protocol/sasl_authenticate.h"
+#include "kafka/protocol/sasl_handshake.h"
+#include "kafka/protocol/types.h"
+#include "kafka/server/handlers/details/security.h"
+#include "model/ktp.h"
+#include "model/timeout_clock.h"
+#include "random/generators.h"
+#include "redpanda/tests/fixture.h"
+#include "security/acl.h"
+#include "security/scram_algorithm.h"
+#include "security/scram_authenticator.h"
+#include "security/types.h"
+#include "types.h"
+
+#include <absl/algorithm/container.h>
+#include <boost/test/tools/old/interface.hpp>
+
+#include <chrono>
+
+static const int32_t not_provided_authz_return = -2147483648;
+static const std::vector<security::acl_operation> default_cluster_auths = {
+  security::acl_operation::create,
+  security::acl_operation::alter,
+  security::acl_operation::describe,
+  security::acl_operation::cluster_action,
+  security::acl_operation::describe_configs,
+  security::acl_operation::alter_configs,
+  security::acl_operation::idempotent_write};
+
+static const std::vector<security::acl_operation> default_topics_auths = {
+  security::acl_operation::read,
+  security::acl_operation::write,
+  security::acl_operation::create,
+  security::acl_operation::describe,
+  security::acl_operation::remove,
+  security::acl_operation::alter,
+  security::acl_operation::describe_configs,
+  security::acl_operation::alter_configs};
+
+static const ss::sstring test_username = "test";
+static const ss::sstring test_acl_principal = "User:test";
+static const ss::sstring test_password = "password";
+
+class metadata_fixture : public redpanda_thread_fixture {
+protected:
+    void create_topic(ss::sstring tp, int32_t partitions, int16_t rf) {
+        kafka::creatable_topic topic;
+        topic.name = model::topic(tp);
+        topic.num_partitions = partitions;
+        topic.replication_factor = rf;
+
+        auto req = kafka::create_topics_request{.data{
+          .topics = {topic},
+          .timeout_ms = 10s,
+          .validate_only = false,
+        }};
+
+        auto client = make_kafka_client().get0();
+        client.connect().get0();
```

Comment:
nitpick, `get0` has been deprecated for years (there are other instances of this, too).
```suggestion
        client.connect().get();
```